src/librustc/mir/interpret/allocation.rs

```diff
@@ -16,15 +16,20 @@ use std::borrow::Cow;
 #[derive(Clone, Debug, Eq, PartialEq, PartialOrd, Ord, Hash, RustcEncodable, RustcDecodable)]
 pub struct Allocation<Tag=(),Extra=()> {
     /// The actual bytes of the allocation.
-    /// Note that the bytes of a pointer represent the offset of the pointer
-    pub bytes: Vec<u8>,
+    /// Note that the bytes of a pointer represent the offset of the pointer. Bytes of undefined
+    /// allocations may be allocated lazily as bytes are initialized but either none or all bytes
+    /// are allocated.
+    crate bytes: Vec<u8>,
     /// Maps from byte addresses to extra data for each pointer.
     /// Only the first byte of a pointer is inserted into the map; i.e.,
     /// every entry in this map applies to `pointer_size` consecutive bytes starting
     /// at the given offset.
     pub relocations: Relocations<Tag>,
-    /// Denotes undefined memory. Reading from undefined memory is forbidden in miri
-    pub undef_mask: UndefMask,
+    /// Denotes which part of this allocation is initialized
+    crate undef_mask: UndefMask,
+    /// The size of the allocation.
+    /// Used after initialiation of an undef region that is not allocated.
```

Comment:
```suggestion
    /// Can be bigger than `bytes.len()` because we do not allocate all bytes immediately.
```

I don't understand what the second sentence here is telling me, is this a correct interpretation.

Reply:
That part of the doc shouldn't yet be there, that's part of the changeset for eliding allocation which was intended as a follow up instead :/